Zend/zend_multiply.h

```diff
@@ -19,10 +19,32 @@
 
 /* $Id$ */
 
+#include "zend_portability.h"
+
 #ifndef ZEND_MULTIPLY_H
 #define ZEND_MULTIPLY_H
 
-#if (defined(__i386__) || defined(__x86_64__)) && defined(__GNUC__)
+#if __has_builtin(__builtin_smull_overflow) && SIZEOF_LONG == SIZEOF_ZEND_LONG
```

Comment:
Both GCC and Clang support this intrinsics. However, because GCC does not support `__has_builtin` it would be better to check whether it exists in autoconf (like we do for a number of other builtin's).